x-pack/plugin/stack/src/javaRestTest/java/org/elasticsearch/xpack/stack/EcsDynamicTemplatesIT.java

```diff
@@ -0,0 +1,400 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.stack;
+
+import org.apache.http.HttpStatus;
+import org.elasticsearch.client.Request;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.network.NetworkAddress;
+import org.elasticsearch.common.time.DateFormatter;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.test.cluster.ElasticsearchCluster;
+import org.elasticsearch.test.rest.ESRestTestCase;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xcontent.json.JsonXContent;
+import org.elasticsearch.xpack.core.template.TemplateUtils;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.instanceOf;
+
+@SuppressWarnings("unchecked")
+public class EcsDynamicTemplatesIT extends ESRestTestCase {
+
+    @ClassRule
+    public static ElasticsearchCluster cluster = ElasticsearchCluster.local().module("mapper-extras").module("wildcard").build();
+
+    // The dynamic templates we test against
+    public static final String ECS_DYNAMIC_TEMPLATES_FILE = "/ecs-dynamic-mappings.json";
+
+    // The current ECS state (branch main) as a containing all fields in flattened form
+    private static final String ECS_FLAT_FILE_URL = "https://raw.githubusercontent.com/elastic/ecs/main/generated/ecs/ecs_flat.yml";
+
+    private static final Set<String> OMIT_FIELD_TYPES = Set.of("object", "flattened", "nested");
```

Comment:
I get why we omit `nested` mappings, but why do we omit `flattened` as well?

Reply:
Good point! I relied on [testecs](https://github.com/elastic/testecs/blob/e4b0837b03d9a63ba6e71f5eb715065fc9050770/ecs.py#L49) on this one, but we should test our [`flattened` type dynamic template](https://github.com/elastic/elasticsearch/blob/735329e1108a085ef36734872244e925cc196a01/x-pack/plugin/core/template-resources/src/main/resources/ecs-dynamic-mappings.json#L177C10-L188) as well!

I will add it to the test, but the test value will always be nested under the flattened fields, otherwise this can't work with `subobjects: false`, where the field names are evaluated as root level, which means they cannot be matched against a `path_match`.

Reply:
@P1llus why wasn't it tested originally? Do you see any reason not to test it here?

Reply:
@eyalkoren No real reason, it was simply that the initial script I made was not meant to cover all usecases, and flattened support was not really in the scope at that point.
It should indeed be tested though.